src/main/kotlin/com/exactpro/th2/codec/xml/XmlPipelineCodec.kt

```diff
@@ -123,9 +176,43 @@ open class XmlPipelineCodec(private val settings: XmlPipelineCodecSettings, xsdM
 
             val msgType: String = pointer?.let { map.getNode<String>(it) } ?: map.keys.first()
 
+            return msgType to map
+        } catch (e: Exception) {
+            throw DecodeException("Can not decode message. Can not parse XML. $xmlString", e)
+        }
+    }
+
+    private fun decodeOneProto(rawMessage: ProtoRawMessage): ProtoMessage {
+        val xmlString = rawMessage.body.toStringUtf8()
+
+        try {
+            val logId = rawMessage.logId
+            validator.validate(rawMessage.body.toByteArray())
+            LOGGER.debug("Validation of incoming raw message complete: $logId")
+
+            val (msgType, map) = decodeOne(rawMessage.body.toByteArray(), xmlString, logId)
```

Comment:
Why convert the body to a byte array twice? Let's extract it into a variable instead